### DISCUSSION TITLE
Procmacro rust-analyzer error

### DISCUSSION THREAD

**Discussion by DoctorRainer** [helix-editor/helix#9928] [Troubleshooting] [answered]
### Summary

I get

proc macro 'main' not expanded: Failed to run proc-macro server from path
/Users/local/.rustup/toolchains/stable-aarch64-apple-darwin/libexec/rust-analyzer-proc-macro-srv,
error: Custom & kind: Other, error: "proc-macro server's api version (3) is newer than
rust-analyzer's (2)" }
(unresolved-proc-macro)

When I'm trying to use #[main::tokio]

### Reproduction Steps

1. Use rustup default stable
2. Create a new project with tokio with #[tokio::main]
3. Get this error

### Platform

macOS

### Terminal Emulator

iTerm

### Installation Method

source

### Helix Version

58022586a00a23fe20d12a563dbf6e97b3e64df0

**Comment by kirawi** [comment]
This is a rust-analyzer issue. You probably have an outdated rust-analyzer installation relative to your rust installation.

**Comment by DoctorRainer** [comment, marked as answer]
Oh, sorry, it's true. For some reason rustup doesn't install appropriate version of rust-analyser so I had to install the binary from their repo

**Comment by calbaker** [comment]
@DoctorRyner , I'm on an M1 ARM Mac and struggling to install the binary.  Is there any chance you know how to do this on Mac?

**Reply by DoctorRainer** [reply]
@calbaker you can find M1 binary here https://github.com/rust-lang/rust-analyzer/releases

I use M1 Max myself

**Comment by calbaker** [comment]
@DoctorRyner , so you download  rust-analyzer-darwin-arm64.vsix?   I assume after that:
1. find the executable in the extracted folder
1. move the executable somewhere on the `PATH` and set the right permissions (which permissions are right for this?) 
1. ...
1.  PROFIT!!!

Could you steer me right wherever I'm wrong here?

**Reply by kirawi** [reply]
Is the file named `rust-analyzer`?

**Reply by DoctorRainer** [reply]
There is .gz file, I don't even know what vsix is. You just rename it to rust-analyzer and put it in the path

**Reply by kirawi** [reply]
Ah, I looked it up. vsix is for Visual Studio. Yeah you want the zip instead.
